sdk/ml/azure-ai-ml/tests/test_utilities/utils.py

```diff
@@ -273,12 +275,22 @@ def delete_file_if_exists(file_path: str):
         os.remove(file_path)
 
 
+@contextmanager
+def skip_sleep_in_lro_polling(work_only_if_not_live=True):
+    if work_only_if_not_live and is_live():
+        yield
+        return
+
+    from azure.core.polling.base_polling import LROBasePolling
+    with patch.object(LROBasePolling, "_sleep", return_value=None):
+        yield
+
+
```

Comment:
we could fixture-ize this as well and make it an autouse fixture?

Reply:
that way each test that uses a poller automatically picks up this logic (but can opt out of it)

Reply:
I have moved this logic to recorded tests as it seems ought to be a default behavior for all recorded tests. Also, we can avoid impact those non-recorded tests in this way. 

Do you think we still need to provide a way to skip this mock in recorded method.

Reply:
Well, it seems that recorded_tests also works for other packages. I'll do a refactor.